domain/src/test/java/org/oppia/android/domain/onboarding/AppStartupStateControllerTest.kt

```diff
@@ -765,6 +782,110 @@ class AppStartupStateControllerTest {
     assertThat(mode.buildFlavorNoticeMode).isEqualTo(NO_NOTICE)
   }
 
+  @Test
+  fun testController_appAndOsDeprecationEnabled_userNotOnboarded_returnsCorrectStartupMode() {
+    executeInPreviousAppInstance { testComponent ->
+      testComponent.getPlatformParameterController().updatePlatformParameterDatabase(
+        listOf(enableAppAndOsDeprecation)
+      )
+      testComponent.getTestCoroutineDispatchers().runCurrent()
+    }
+    setUpDefaultTestApplicationComponent()
+
+    monitorFactory.ensureDataProviderExecutes(platformParameterController.getParameterDatabase())
+    testCoroutineDispatchers.runCurrent()
+
+    val appStartupState = appStartupStateController.getAppStartupState()
+
+    val startupMode = monitorFactory.waitForNextSuccessfulResult(appStartupState)
+    assertThat(startupMode.startupMode).isEqualTo(USER_NOT_YET_ONBOARDED)
+  }
+
+  @Test
+  fun testController_appAndOsDeprecationEnabled_userIsOnboarded_returnsCorrectStartupMode() {
```

Comment:
Maybe replace with what the correct expected `CorrectStartupMode` is, in case this changes in the future.